File: Bedu/Java/Clase5/FragmentPermisos/app/src/main/java/com/charlye934/fragmentpermisos/MainActivity.java
package com.charlye934.fragmentpermisos;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.Manifest;
import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.Switch;
import android.widget.Toast;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {

    Map<String,String> permisos = new HashMap<>();
    String tipo[] = {"camara","mensaje","contactos","calendario"};
    String tipoPermiso[] = {Manifest.permission.CAMERA,Manifest.permission.READ_SMS,Manifest.permission.READ_CONTACTS, Manifest.permission.READ_CALENDAR};

     @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

         for(int i=0; i<tipo.length; i++)
             permisos.put(tipo[i],tipoPermiso[i]);
    }

    public void Cambio(View view){
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();

        String tag = String.valueOf(view.getTag());
        Fragment fragment = null;

        if(Permisos(tag)){
            switch(tag){
                case "camara":
                    fragment = new CamaraFragment();
                    break;
                case "contactos":
                    fragment = new ContactosFragment();
                    break;
                case "mensaje":
                    fragment = new MensajesFragment();
                    break;
                case "calendario":
                    fragment = new CalendarioFragment();
                    break;
            }
            fragmentTransaction.add(R.id.contenedor, fragment, "f1");
            fragmentTransaction.addToBackStack("f1");
            fragmentTransaction.commit();
        }
    }

    public boolean Permisos(String tag){
        int version = Build.VERSION.SDK_INT;

        if(version >= Build.VERSION_CODES.M){
            if (ContextCompat.checkSelfPermission(getApplicationContext(), permisos.get(tag)) != PackageManager.PERMISSION_GRANTED) {
                ActivityCompat.requestPermissions(MainActivity.this, new String[]{permisos.get(tag)}, 1000);
                return false;
            }else
                return true;
        }else{
            Toast.makeText(getApplicationContext(),"No es necesario pedir permisos",Toast.LENGTH_SHORT).show();
            return true;
        }
    }
}

File: Bedu/kotlin/Clase1-5/FragmentPermisosKotlin/app/src/main/java/com/example/fragmentpermisoskotlin/MainActivity.kt
package com.example.fragmentpermisoskotlin

import android.Manifest
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var permisos: Map<String,String>  = mapOf("camara" to  Manifest.permission.CAMERA, "mensaje" to Manifest.permission.READ_SMS, "contactos" to Manifest.permission.READ_CONTACTS, "calendario" to Manifest.permission.READ_CALENDAR)
    var fragment: Map<String,Fragment> = mapOf("camara" to CamaraFragment(), "mensaje" to MensajesFragment(), "contactos" to ContactosFragment(), "calendario" to CalendarioFragment())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    fun Cambio(view:View){
        //contenedor.removeAllViews()
        val fragmentManager = supportFragmentManager
        var fragmentTransaction = fragmentManager.beginTransaction()
        var tag = view.getTag().toString()
        if(permisos(tag)){
            fragmentTransaction.remove(fragment[tag]!!)
            fragmentTransaction.add(R.id.contenedor, fragment[tag]!!,"f1")
            fragmentTransaction.addToBackStack("f1")
            fragmentTransaction.commit()
        }
    }

    fun permisos(tag:String):Boolean{
        var version = Build.VERSION.SDK_INT

        if(version>=Build.VERSION_CODES.M){
            if(ContextCompat.checkSelfPermission(applicationContext, permisos.get(tag).toString()) != PackageManager.PERMISSION_GRANTED){
                ActivityCompat.requestPermissions(this@MainActivity, arrayOf(permisos[tag]), 1000)
                return false
            }else{ return true }
        } else {
            Toast.makeText(applicationContext, "No es necesario pedir permisos", Toast.LENGTH_SHORT).show()
            return true
        }
    }
}
